app/controllers/admin/collections_controller.rb

```diff
@@ -252,24 +252,21 @@ def attach_poster
         @collection.poster.content = resized_poster
         @collection.poster.mime_type = 'image/png'
         @collection.poster.original_name = poster_file.original_filename
-        flash[:success] = "Poster file succesfully added."
+        flash[:success] = "Poster file successfully added."
       else
         flash[:error] = "Uploaded file is not a recognized poster image file"
       end
     else
       @collection.poster.content = ''
       @collection.poster.original_name = ''
-      flash[:success] = "Poster file succesfully removed."
+      flash[:success] = "Poster file successfully removed."
+      redirect_to admin_collection_path(@collection)
     end
 
-    unless @collection.save
-      flash[:success] = nil
-      flash[:error] = "There was a problem storing the poster image."
-    end
-
-    respond_to do |format|
-      format.html { redirect_to admin_collection_path(@collection) }
-    end
+    return if @collection.save
+    flash[:success] = nil
+    flash[:error] = "There was a problem storing the poster image."
+    redirect_to admin_collection_path(@collection)
```

Comment:
It's not exactly clear to me which cases should be redirecting and which shouldn't.  Shouldn't they all?  Or maybe the flash message rendering is dependent on this not being a redirect?

Reply:
It was my understanding that a redirect is necessary for the flash messages to render on the page. Maybe we can put this at the end of the method like it was before?